### DISCUSSION TITLE
Artist Images from Remote Share

### DISCUSSION THREAD

**Discussion by rh8056** [music-assistant/support#2630] [Q&A] [answered]
I'm just now diving into Music Assistant, but so far really liking what I'm seeing. I have a file structure like this on a remote nas:

```
/music/albums
/music/coverart/Albums
/music/coverart/Artists
/music/other_stuff_that_i_dont_want_ma_to_read
```

I have two SMB mounts for `/albums` and `/coverart`. I added both of these into MA thinking that it would scan my Artists folder and use those images for the Artist images page. Images in that folder are named:

```
$.../coverart/Artists# ls -all A*
-rw-r--r-- 1 nobody nogroup   41082 Aug 13  2021 'A Blaze of Feather.jpg'
-rw-r--r-- 1 nobody nogroup  477488 Mar 10  2021 'A Fine Frenzy.png'
-rw-r--r-- 1 nobody nogroup  344604 Mar 10  2021 'A Great Big World.png'
-rw-r--r-- 1 nobody nogroup  243653 Mar 10  2021  Adele.jpg
-rw-r--r-- 1 nobody nogroup   82811 Mar 10  2021  Aerosmith.jpg
-rw-r--r-- 1 nobody nogroup   32621 Aug 13  2021 'Alexi Murdoch.jpg'
-rw-r--r-- 1 nobody nogroup   40262 Aug 13  2021 'Alice Cooper.jpg'
-rw-r--r-- 1 nobody nogroup   27937 Feb 10  2022 'Alison Krauss & Robert Plant.jpg'
-rw-r--r-- 1 nobody nogroup   10396 Mar 10  2021 'Alison Krauss & The Cox Family.jpg'
-rw-r--r-- 1 nobody nogroup   79112 Mar 10  2021 'Alison Krauss & Union Station.jpg'
-rw-r--r-- 1 nobody nogroup  739280 Mar 10  2021 'Alison Krauss.jpg'
etc etc...
```

However, I'm not seeing these in my Artist view of MA:

![image](https://github.com/user-attachments/assets/5e65bb06-7b54-4d7a-8550-26afe6c12f87)

The only artists showing images are those that I clicked into to see what the views were like, which I understand is the lazy loading aspect, and the images are from online metadata, not the files on my NAS. Do I have this configured incorrectly? I thought my existing files followed the notes here:

https://music-assistant.io/music-providers/filesystem/#known-issues-notes

It doesn't really specify anything about the directory hierarchy, so perhaps what I have needs to be changed. Is there another way this should be configured? Thanks!

**Comment by OzGav** [comment]
The docs do specify 

> Music Assistant has support for both embedded artwork and artwork stored in a common folder structure of Artist \ Album

Then further down you are directed to https://kodi.wiki/view/Music_tagging which shows a diagram of what the directory structure should look like

**Reply by rh8056** [reply]
I'm not understanding how that diagram specifies where artist artwork should be stored. Do I need to have a folder per artist, and then a file per artist?

EDIT: Perhaps I should also clarify that my `/albums/` folder looks like this:

```
.../music/albums# ls -all -d A*
drwxr-xr-x 2 nobody nogroup 16 Jun 22  2023 'A Blaze of Feather - 2017 - A Blaze of Feather'
drwxr-xr-x 2 nobody nogroup  9 Jun 22  2023 'A Blaze of Feather - 2017 - EP 1'
drwxr-xr-x 2 nobody nogroup 13 Jun 22  2023 'A Blaze of Feather - 2020 - Labyrinth'
drwxr-xr-x 2 nobody nogroup 20 Jun 22  2023 'A Fine Frenzy - 2007 - One Cell in the Sea'
drwxr-xr-x 2 nobody nogroup 19 Jun 22  2023 'A Great Big World - 2014 - Is There Anybody Out There'
drwxr-xr-x 2 nobody nogroup 17 Jun 22  2023 'Adele - 2008 - 19'
drwxr-xr-x 3 nobody nogroup 16 Jun 22  2023 'Adele - 2011 - 21'
drwxr-xr-x 2 nobody nogroup 19 Jun 22  2023 'Adele - 2015 - 25'
drwxr-xr-x 2 nobody nogroup 15 Jun 22  2023 'Adele - 2021 - 30'
```

I do not have all Adele albums in one folder, for example. Thanks!

**Reply by OzGav** [reply, marked as answer]
In order to have your local antwork appear you need a folder per artist and in the folder you will have image files for the artist thumb and fanart. Below that you will have a folder per album and in those you will have images for the album cover.

**Reply by rh8056** [reply]
Shouldn't this be possible to set up in another share then? I attempted the following directory structure when trying to figure it out. My thought was that MA matches information across sources, so I was thinking it might get the artist images from the `/coverart/` share, and the media from the `/albums/` share:

```
/music/coverart/A Blaze of Feather/artist.jpg
/music/coverart/A Blaze of Feather/A Blaze of Feather.jpg
/music/coverart/A Blaze of Feather/fanart.jpg
```

I have `/music/coverart` mounted as a separate remote share to MA, so I would have thought MA would have indexed `A Blaze of Feather/artist.jpg` and used that for the artist image in the view. I do not want to rearrange all of my media files in the structure of an Artist then Album because I have many classical albums that do not lend themselves to that structure very well. Thanks!

**Reply by OzGav** [reply]
You don’t have to do that for your classical albums that is also shown on the kodi page.

**Reply by rh8056** [reply]
Sorry if I'm being dense, I'm just trying to figure out how to work with local art, without having to rearrange all of my media. Is there a hard requirement that the artist art needs to be along side the media files? Can a user configure a separate mount for artist (not album) art?

This is what I have now:

```
2024-07-16 07:04:34.422 INFO (MainThread) [music_assistant.providers.filesystem_smb] Mounting //192.168.3.204/MusicAssistant to /tmp/filesystem_smb--Xpz6yu4G
2024-07-16 07:04:34.423 INFO (MainThread) [music_assistant.providers.filesystem_smb] Mounting //192.168.3.204/MusicAssistantArt to /tmp/filesystem_smb--PvGVfwzC
```

From another device, browsing that MusicAssistantArt SMB:

```
$ smbclient '\\192.168.3.204\MusicAssistantArt' -U guest
Enter WORKGROUP\guest's password: 
Try "help" to get a list of possible commands.
smb: \> ls
  .                                   D        0  Mon Jul 15 17:01:57 2024
  ..                                  D        0  Sun Feb 18 17:52:16 2024
  A Blaze of Feather                  D        0  Mon Jul 15 17:12:43 2024
  Composers                           D        0  Thu Jun 22 14:13:55 2023
  Albums                              D        0  Thu Jun 22 14:24:05 2023
  Artists                             D        0  Mon Jul 15 14:15:48 2024

		65388221184 blocks of size 1024. 27176183296 blocks available
smb: \> cd "A Blaze of Feather"
smb: \A Blaze of Feather\> ls
  .                                   D        0  Mon Jul 15 17:12:43 2024
  ..                                  D        0  Mon Jul 15 17:01:57 2024
  fanart.jpg                          N    41082  Mon Jul 15 17:12:43 2024
  artist.jpg                          N    41082  Mon Jul 15 16:59:27 2024

		65388221184 blocks of size 1024. 27176183296 blocks available
smb: \A Blaze of Feather\> 
```

As I read the MA documentation:

> Use the following naming convention for local artwork (png should also work). Artist thumb: artist.jpg; Album thumb: folder.jpg or cover.jpg; Fan Art (used as background in banners): fanart.jpg. Artist thumb and Fanart should be in the folder with the artist name. 

I have a folder named "A Blaze of Feather" and a file in that folder named artist.jpg. Should this work?

**Reply by marcelveldt** [reply]
Yes that is a hard requirement, just like many other media software. Follow the folder structure like Kodi explained very well in their docs and you're all set.

**Reply by rh8056** [reply]
Ok, that's too bad. I currently use a media software program (JRiver), that does not require that. And this line in the Kodi documentation seems to suggest the folder structure isn't a requirement:

> Kodi is quite flexible with the folder structure. The diagram below is an example. You should group your albums in a manner that makes sense to you.
